Tests/Approval.Tests/ApprovedApi/FluentAssertions/net47.verified.txt

```diff
@@ -647,6 +647,20 @@ namespace FluentAssertions.Common
     {
         System.TimeSpan Elapsed { get; }
     }
+    public class ReadOnlyNonGenericCollectionWrapper<TCollection, TItem> : System.Collections.Generic.ICollection<TItem>, System.Collections.Generic.IEnumerable<TItem>, System.Collections.IEnumerable
```

Comment:
What members do you need access to?
[Hyrum's Law](https://www.hyrumslaw.com/) applies to everything we expose.

At least I'd like to change the interface methods to explicit interface implementations instead.

Reply:
The proposed extension library needs to be identify the type, that's all. Explicit interface implementations should work just fine.

Reply:
Oh, I'm mistaken, the `UnderlyingCollection` property is needed as well.

Should I just make that one property `public` or would you like another interface to be made to expose that property so that it can be accessible but not visible by default?

Reply:
Hmm, in fact, if an interface were introduced for `UnderlyingCollection`, then the consuming code would only need _that_, not the concrete type, so that can go back to being `internal`.

Reply:
Okay the implementation has been switched to this. The concrete type is not exposed at all.